app/src/main/java/io/stormbird/wallet/service/OpenseaService.java

```diff
@@ -62,20 +62,24 @@ public OpenseaService(Context ctx) {
         for (int i = 0; i < assets.length(); i++)
         {
             Asset asset = new Gson().fromJson(assets.getJSONObject(i).toString(), Asset.class);
-
-            Token token = foundTokens.get(asset.getAssetContract().getAddress());
-            if (token == null)
+            if (asset != null && (asset.getAssetContract().getSchemaName() == null
+                                || asset.getAssetContract().getSchemaName().length() == 0
+                                || asset.getAssetContract().getSchemaName().contains("721"))) //filter ERC721
```

Comment:
Looks good, except is it safer to check that it's "ERC721" or somehow that it doesn't match something like "ERC2721"?